user/internal/repository/repository_user.go

```diff
@@ -10,3 +12,11 @@ func (r *Repository) GetUserById(id string) (*models.User, error) {
 	}
 	return &user, nil
 }
+
+func (r *Repository) CreateUser(user *models.User) error {
```

Comment:
Should you construct the new struct instead of using model.User? In this case, that mean, we can also specify id when creating user?

Reply:
why new struct tho when it has the same fields

Reply:
though i could allow specific ids, and default to random if not provided instead

Reply:
Ok

Reply:
I think that the service should not be able to define id by itself. For example, assume 1 service by 2 instance, the first instance randomly generate uuid as A and the second also randomly generate uuid as A as well. They call createUser at the same time. One of them might get the error due to PK constraint but if we use uuid from Postgres this case will not happen (It do only one transaction at the time).

TL;DR my point is omitting id from the parameter of createUser

Reply:
oh, i thought u meant u want to specify id

ok

Reply:
it's a little hard to create a new type from models.User without the Id part tho

But isn't it normal anyways? that problem only occurs if the code specifies the Id when creating user, and we're not doing that.

Other cases of gorm also use normal models, even though it has ID field on it.